src/odemis/acq/stream/_sync.py

```diff
@@ -1806,6 +1994,17 @@ def _runAcquisition(self, future):
                         raise CancelledError()
                     s._dataflow.unsubscribe(self._subscribers[i])
 
+                px_x, px_y = divmod(spots_sum, rep[0])
+                px_idx = (px_x, px_y)
```

Comment:
That should be merged with px_idx computed at the beginning. (You could just drop these two lines, or use divmod up).